1.32.0: Testing fails without `HOME` environment variable

When tests are run without the `HOME` environment variable then they fails:
```
__________ CommandLineTestCase.test_run_with_user_global_config_file ___________

self = <tests.test_cli.CommandLineTestCase testMethod=test_run_with_user_global_config_file>

    @unittest.skipIf(os.environ.get('GITHUB_RUN_ID'), '$HOME not overridable')
    def test_run_with_user_global_config_file(self):
        home = os.path.join(self.wd, 'fake-home')
        dir = os.path.join(home, '.config', 'yamllint')
        os.makedirs(dir)
        config = os.path.join(dir, 'config')
    
>       self.addCleanup(os.environ.update, HOME=os.environ['HOME'])

tests/test_cli.py:316: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = environ({'LC_CTYPE': 'C.UTF-8', 'PYTHONPATH': '...', 'PYTEST_CURRENT_TEST': 'tests/test_cli.py::CommandLineTestCase::test_run_with_user_global_config_file (call)'})
key = 'HOME'

    def __getitem__(self, key):
        try:
            value = self._data[self.encodekey(key)]
        except KeyError:
            # raise KeyError with the original key value
>           raise KeyError(key) from None
E           KeyError: 'HOME'

/usr/lib/python3.9/os.py:679: KeyError
```

--- a/tests/test_cli.py
+++ b/tests/test_cli.py
@@ -328,7 +328,6 @@ class CommandLineTestCase(unittest.TestCase):
             cli.run(('-c', f.name, os.path.join(self.wd, 'a.yaml')))
         self.assertEqual(ctx.returncode, 1)
 
-    @unittest.skipIf(os.environ.get('GITHUB_RUN_ID'), '$HOME not overridable')
     def test_run_with_user_global_config_file(self):
         home = os.path.join(self.wd, 'fake-home')
         dir = os.path.join(home, '.config', 'yamllint')

--- a/tests/test_cli.py
+++ b/tests/test_cli.py
@@ -315,7 +315,7 @@ class CommandLineTestCase(unittest.TestCase):
         os.makedirs(dir)
         config = os.path.join(dir, 'config')
 
-        self.addCleanup(os.environ.update, HOME=os.environ['HOME'])
+        self.addCleanup(os.environ.__delitem__, 'HOME')
         os.environ['HOME'] = home
 
         with open(config, 'w') as f:

--- a/tests/test_cli.py
+++ b/tests/test_cli.py
@@ -66,6 +66,26 @@ def utf8_available():
         return False
 
 
+def setUpModule():
+    # yamllint uses these environment variables to find a config file.
+    env_vars_that_could_interfere = (
+        'YAMLLINT_CONFIG_FILE',
+        'XDG_CONFIG_HOME',
+        # These variables are used to determine where the user’s home
+        # directory is. See
+        # https://docs.python.org/3/library/os.path.html#os.path.expanduser
+        'HOME',
+        'USERPROFILE',
+        'HOMEPATH',
+        'HOMEDRIVE'
+    )
+    for name in env_vars_that_could_interfere:
+        try:
+            del os.environ[name]
+        except KeyError:
+            pass
+
+
 class CommandLineTestCase(unittest.TestCase):
     @classmethod
     def setUpClass(cls):

